test/e2e/containerruntimeconfig_test.go

```diff
@@ -0,0 +1,113 @@
+package e2e_test
+
+import (
+	"fmt"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	corev1 "k8s.io/api/core/v1"
+	"github.com/openshift/machine-config-operator/test/e2e/framework"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func TestContainerRuntimeConfigPidsLimit(t *testing.T) {
+	cs := framework.NewClientSet("")
+	testUUID := uuid.NewUUID()
+
+	matchValue := fmt.Sprintf("pids-limit-%s", testUUID)
+	mcpKey := "machine-config-pool/test"
+	ctrcfgName := fmt.Sprintf("ctrcfg-%s", testUUID)
+
+	_, err := exec.Command("oc", "label", "mcp", "worker", mcpKey+"="+matchValue).CombinedOutput()
+	require.Nil(t, err, "unable to label worker mcp")
+
+	defer func() {
+		_, err := exec.Command("oc", "label", "mcp", "worker", mcpKey+"-").CombinedOutput()
+		require.Nil(t, err, "unable to remove label worker mcp")
+	}()
+
+	cfg := &mcfgv1.ContainerRuntimeConfiguration{
+		PidsLimit: 2048,
+	}
+
+	cleanupCRC := createCRCWithConfig(t, cs, ctrcfgName, mcpKey, matchValue, cfg)
+	defer cleanupCRC()
+
+	mcName, err := waitForRenderedConfigFromCtrcfg(t, cs, "worker", ctrcfgName)
+	require.Nil(t, err, "failed to render machine config from container runtime config")
+
+	err = waitForPoolComplete(t, cs, "worker", mcName)
+	require.Nil(t, err)
+}
+
+func createCRCWithConfig(t *testing.T, cs *framework.ClientSet, name, key, value string, config *mcfgv1.ContainerRuntimeConfiguration) func() {
+	ctrcfg := &mcfgv1.ContainerRuntimeConfig{}
+	ctrcfg.ObjectMeta = metav1.ObjectMeta{
+		Name: name,
+	}
+	spec := mcfgv1.ContainerRuntimeConfigSpec{
+		MachineConfigPoolSelector: &metav1.LabelSelector{
+			MatchLabels: make(map[string]string),
+		},
+		ContainerRuntimeConfig: config,
+	}
+	spec.MachineConfigPoolSelector.MatchLabels[key] = value
+	ctrcfg.Spec = spec
+
+	_, err := cs.ContainerRuntimeConfigs().Create(ctrcfg)
+	require.Nil(t, err)
+	return func() {
+		cs.ContainerRuntimeConfigs().Delete(name, &metav1.DeleteOptions{})
```

Comment:
why I love how this all works it's kinda suboptimal as we're ignoring the error - maybe handle it somehow?

Reply:
Do you think the test should fail if it fails in cleaning up? Or is logging it sufficient

Reply:
I think this should be a continuation of the actual test where you make sure that rolling back (deleting) the ctrcfg brings the cluster back to normal (see #1412 for instance). TLDR: it should fail

Reply:
Sgtm

Reply:
also still not sure why tests failed 🤔looks super weird but this is OT from this convo